src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -180,6 +178,15 @@ public override void OnCustom(string name, Activity activity, object payload)
                                     break;
                             }
                         }
+
+                        try
+                        {
+                            this.options.EnrichWithIDbCommand?.Invoke(activity, command as IDbCommand);
```

Comment:
Seeing the `as IDbCommand` here is a bit strange. Would it ever be something else? Is user expected to handle null?

Depending on the answers, might be able to change this...

```csharp
private readonly PropertyFetcher<object> commandFetcher = new PropertyFetcher<object>("Command");
```

...to...

```csharp
private readonly PropertyFetcher<IDbCommand> commandFetcher = new PropertyFetcher<IDbCommand>("Command");
```

Or perhaps skip the call...

```csharp
if (command is IDbCommand typedCommand)
{
  this.options.EnrichWithIDbCommand?.Invoke(activity, typedCommand);
}
```

Reply:
Just looking at this, maybe moving it just after the braces for switch command makes more sense. Then would be guaranteed to be some kind of a commandtype.

Reply:
Done, how does that look look?